### PR TITLE
[FLINK-18055] [sql-client] Fix catalog/database does not exist in sql client

### PR DESCRIPTION
**Pull request by godfreyhe**
## What is the purpose of the change

*When executing `use catalog hive` in sql client, CatalogException will occur. The reason is SqlCommandParser adds `` for catalog name, which is unnecessary. similar case for use database command *

## Brief change log

  - *ignore `` for catalog name in SqlCommandParser*
  - *only return database name for use database command in SqlCommandParser*

## Verifying this change

This change added tests and can be verified as follows:

  - *Update existing test: SqlCommandParserTest*
  - *Added test methods in CliClientTest to for e2e test*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/cli/CliClientTest.java [hunk 1]
```diff
@@ -166,6 +167,64 @@ public void write(int b) throws IOException {
...
+						throw new SqlExecutionException("unexpected database name: " + databaseName);
+					}
+				})
+				.build();
+
+		InputStream inputStream = new ByteArrayInputStream("use db;\n".getBytes());
+		ByteArrayOutputStream outputStream = new ByteArrayOutputStream(256);
+		CliClient cliClient = null;
+		SessionContext sessionContext = new SessionContext("test-session", new Environment());
+		String sessionId = executor.openSession(sessionContext);
```
Comment (leonardBang) [inline flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/cli/CliClientTest.java:213]: Could we extract this code to a function for reuse purpose?

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/SqlCommandParser.java [hunk 2]
```diff
@@ -132,11 +132,10 @@ private SqlCommandParser() {
 			cmd = SqlCommand.DROP_CATALOG;
 		} else if (operation instanceof UseCatalogOperation) {
 			cmd = SqlCommand.USE_CATALOG;
-			operands = new String[] { String.format("`%s`", ((UseCatalogOperation) operation).getCatalogName()) };
+			operands = new String[] { ((UseCatalogOperation) operation).getCatalogName() };
```
Comment (jxeditor) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/SqlCommandParser.java:135]:
```suggestion
			operands = new String[]{StringUtils.strip(((UseCatalogOperation) operation).getCatalogName(), "`")};
```

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/SqlCommandParser.java [hunk 3]
```diff
@@ -132,11 +132,10 @@ private SqlCommandParser() {
...
 		} else if (operation instanceof UseDatabaseOperation) {
 			cmd = SqlCommand.USE;
-			UseDatabaseOperation op = ((UseDatabaseOperation) operation);
-			operands = new String[] { String.format("`%s`.`%s`", op.getCatalogName(), op.getDatabaseName()) };
+			operands = new String[] { ((UseDatabaseOperation) operation).getDatabaseName() };
```
Comment (jxeditor) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/SqlCommandParser.java:138]:
```suggestion
			operands = new String[]{String.format("%s.%s", StringUtils.strip(op.getCatalogName(), "`"), StringUtils.strip(op.getDatabaseName(), "`"))};
```

File: flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/cli/CliClientTest.java [hunk 4]
```diff
@@ -166,6 +167,64 @@ public void write(int b) throws IOException {
...
+				})
+				.build();
+
+		InputStream inputStream = new ByteArrayInputStream("use catalog cat;\n".getBytes());
+		ByteArrayOutputStream outputStream = new ByteArrayOutputStream(256);
+		CliClient cliClient = null;
+		SessionContext sessionContext = new SessionContext("test-session", new Environment());
+		String sessionId = executor.openSession(sessionContext);
+
+		try (Terminal terminal = new DumbTerminal(inputStream, outputStream)) {
```
[the remark this answers is not in the document]
Comment (godfreyhe) [reply flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/cli/CliClientTest.java:186]: `CliClient` does not inherit from `AutoCloseable`, so we can put `CliClient` into `try-with-resource` block

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/SqlCommandParser.java (hunk 2 above)
Comment (godfreyhe) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/SqlCommandParser.java:135]: The quoting string ``` is skipped by sql parser, so we need not to handle this. see the test (`TestItem.validSql("use `default`;", SqlCommand.USE, "default")`) in `SqlCommandParserTest#testCommands`

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/SqlCommandParser.java (hunk 3 above)
Comment (godfreyhe) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/SqlCommandParser.java:138]: catalog name is unnecessary, only database is needed.

**Comment by leonardBang** [comment]
LGTM, +1 on azure green

**Pull request merged**